apps/wallet-ui/src/app/[ecosystem]/page.tsx

```diff
@@ -0,0 +1,17 @@
+"use client";
+import { client } from "@/lib/client";
+import { useRouter } from "next/navigation";
+import { ConnectButton, useActiveAccount } from "thirdweb/react";
+
+export default function Page() {
```

Comment:
do users *have* to be logged in to see this page? (SIWE auth) - if so instead of doing this logic client side simply do this logic based on auth cookie -> redirect server side

Reply:
you can do this in middleware or in this page specifically

Reply:
Good call, we can get their wallet address from the token

Reply:
yep!

Reply:
please still do this